src/components/10-atoms/text/index.js

```diff
@@ -28,38 +31,51 @@ class AXAText extends NoShadowDOM {
     applyDefaults(this);
   }
 
-  render() {
-    // eslint-disable-next-line prefer-destructuring
-    const variant = this.variant;
-    const isSize2 = variant.indexOf('size-2');
-    const isSize3 = variant.indexOf('size-3');
-    const isBold = variant.indexOf('bold');
-
-    const classes = ['a-text'];
-    if (isSize2 === 0) {
-      classes.push('a-text--size-2');
-    }
-    if (isSize3 === 0) {
-      classes.push('a-text--size-3');
+  watch(mode) {
+    switch (mode) {
+      case 'stop':
+        if (this._observer) {
+          this._observer.disconnect();
+        }
+        break;
+      case 'start':
+        if (!this._observer) {
+          this._observer = new MutationObserver(() => this.render());
+        }
+        this._observer.observe(this, {
+          childList: true,
+        });
+        break;
+      default:
+        break;
     }
-    if (isBold === 0) {
-      classes.push('a-text--bold');
-    }
-
-    const { firstChild, firstElementChild } = this;
+  }
 
-    // nodeType === 3 is Text
-    if (firstChild && firstChild.nodeType === 3 && firstElementChild === null) {
-      const p = document.createElement('p');
-      p.innerHTML = firstChild.textContent;
-      p.className = classes.join(' ');
+  render() {
+    // do we have 'pure', non-empty text, e.g. <axa-text>Hello, World</axa-text>?
+    const nonWhitespaceTextNodes = [...this.childNodes].filter(
+      node => node instanceof Text && node.textContent.trim()
+    );
+    if (nonWhitespaceTextNodes.length) {
+      // yes, wrap it in 1 <p> node to keep screenreaders happy (they offer
+      // paragraph-to-paragraph-jumping keyboard shortcuts that work by detecting <p>s):
+      const paragraph = document.createElement('p');
+      paragraph.textContent = this.textContent;
+      // pause child-update watcher
+      this.watch('stop');
+      // delete the children, now that their content has been harvested
       this.innerHTML = '';
```

Comment:
@markus-walther this will disable live dom effect. Is it not better if we keep dom life?

Reply:
This is copied from your original code, so I'm a bit puzzled.

Now we have a MutationObserver, which takes care of any 'live' updating.

As discussed, the new summary assignment of `this.textContent` captures also the mixed case of some text nodes interspersed with some real DOM nodes (cf. https://developer.mozilla.org/de/docs/Web/API/Node/textContent).

Reply:
ok thanks